app/Controllers/AdminDashboard.php

```diff
@@ -235,7 +247,215 @@ public function createSocialMediaType(): ResponseInterface
+        $roomId = $room['room_id'];
+        $password = $room['password'];
+        $isRoomVisible = ($room['visible_from'] !== null && date($room['visible_from']) <= date('Y-m-d H:i:s'));
+
+        $userIds = $this->gatherUserIdsForVideoRoom($eventId);
+
+        $speakerModel = model(SpeakerModel::class);
+        $speakers = [];
+        foreach ($userIds as $userId) {
+            $speakers[$userId] = $speakerModel->getLatestApprovedForEvent($userId, $eventId);
+        }
+
+        usort($speakers, function ($a, $b) {
+            return $a['name'] <=> $b['name'];
+        });
+
+        $speakerLinks = [];
+        foreach ($speakers as $speaker) {
+            $links = ['name' => $speaker['name']];
+            foreach (VideoLinkType::cases() as $linkType) {
+                foreach (VideoSourceType::cases() as $sourceType) {
+                    $links["{$linkType->value}_{$sourceType->value}"] =
+                        VideoRoomHelper::createVideoLink(
+                            $baseUrl,
+                            $roomId,
+                            $password,
+                            $eventId,
+                            $speaker['user_id'],
+                            $speaker['name'],
+                            $linkType,
+                            $sourceType
+                        );
+
+                }
+            }
+            $speakerLinks[] = $links;
+        }
+
+        $result = [
+            'is_visible' => $isRoomVisible,
+            'director' => $this->createDirectorLink($baseUrl, $roomId, $password),
+            'speakers' => $speakerLinks,
+        ];
+
+        return $this->response->setJSON($result);
+    }
+
+    public function setVideoRoomVisible(int $eventId): ResponseInterface
+    {
+        $eventModel = model(EventModel::class);
+        $event = $eventModel->get($eventId);
+        if ($event === null) {
+            return $this
+                ->response
+                ->setJSON(['error' => 'EVENT_NOT_FOUND'])
+                ->setStatusCode(ResponseInterface::HTTP_NOT_FOUND);
+        }
+
+        $videoRoomModel = model(VideoRoomModel::class);
+        if ($videoRoomModel->get($eventId) === null) {
+            return $this
+                ->response
+                ->setJSON(['error' => 'VIDEO_ROOM_DOES_NOT_EXIST'])
+                ->setStatusCode(ResponseInterface::HTTP_NOT_FOUND);
+        }
+
+        $userIds = $this->gatherUserIdsForVideoRoom($eventId);
+
+        $accountModel = model(AccountModel::class);
+        $accounts = $accountModel->getAccounts($userIds);
+
+        // TODO: Decide whether this should be checked or not. If it is checked,
+        //       then the data of 2024 won't work.
+        /*if (count($accounts) != count($userIds)) {
+            return $this
+                ->response
+                ->setJSON(['error' => 'ACCOUNT_NOT_FOUND'])
+                ->setStatusCode(ResponseInterface::HTTP_INTERNAL_SERVER_ERROR);
```

Comment:
It should not be a Problem when 2024 is not working since the event is long gone.

Reply:
Okay, then I will delete those lines. Please re-approve the changes after that :)